src/NLog/Config/NLogXmlElement.cs

```diff
@@ -212,19 +221,59 @@ public void AssertName(params string[] allowedNames)
             throw new InvalidOperationException("Assertion failed. Expected element name '" + string.Join("|", allowedNames) + "', actual: '" + this.LocalName + "'.");
         }
 
+        /// <summary>
+        /// Returns all errors occured during xml parsing.
+        /// If there is no errors - returns null.
+        /// </summary>
+        public string GetParsingErrors()
+        {
+            var errors = GetParsingErrorsInternal().ToArray();
+            if (errors.Length == 0) return null;
+            return string.Join(Environment.NewLine, errors);
+        }
+
+        /// <summary>
+        /// Returns all parsing errors from current and all child elements.
+        /// </summary>
+        private IEnumerable<string> GetParsingErrorsInternal()
```

Comment:
Thanks for the changes!

As from our part, we find it important to have a good public API. I think it would be better API if I could read the details, so that the errors aren't joined to a single string. Do you agree?

The easiest way is to make `GetParsingErrorsInternal()` public and rename it to `GetParsingErrors()` (and drop the `GetParsingErrors()`). What do you think?